src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -37199,20 +37341,32 @@ static bool TryGetValue2([NotNullWhen(true)] out TYPE y)
         return y != null;
     }
 
+    static bool TryGetValue2B([NotNullWhen(true)] out TYPE y)
+    {
+        y = null;
+        return y == null; // 2
```

Comment:
The fact that these work but putting them inside a `||` does not is _really_ confusing for me. I don't think this will be obvious at all for users.

Reply:
Agreed. Still, in terms of maximizing useful warnings and minimizing bad ones, this design seems the best option so far...

---
In reply to: [503605817](https://github.com/dotnet/roslyn/pull/48425#discussion_r503605817) [](ancestors = 503605817)